world.js

```diff
@@ -294,7 +295,16 @@ const _bindHitTracker = app => {
   hitTracker.bind(app);
 
   const die = () => {
-    world.appManager.removeTrackedApp(app.instanceId);
+    if (app.npcPlayer?.avatar) {
+      physicsManager.disableGeometry(app.npcPlayer.characterController);
+      physicsManager.disableGeometryQueries(app.npcPlayer.characterController);
+      app.npcPlayer.avatar.ragdoll = true;
```

Comment:
Definitely should not set this avatar internal state here. This code belongs to the avatar.